addons/mail/tests/test_mail_template.py

```diff
@@ -70,12 +70,57 @@ def test_mail_template_acl(self):
         with self.assertRaises(AccessError):
             self.env['mail.template'].with_user(self.user_employee).create({'body_html': '<p t-esc="\'foo\'"></p>'})
 
+        # Standard employee cannot edit templates from another user, non-dynamic and dynamic
+        with self.assertRaises(AccessError):
+            mail_template.with_user(self.user_employee).body_html = '<p>foo</p>'
         with self.assertRaises(AccessError):
             mail_template.with_user(self.user_employee).body_html = '<p t-esc="\'foo\'"></p>'
 
+        # Standard employee can edit his own templates if not dynamic
+        employee_template.with_user(self.user_employee).body_html = '<p>foo</p>'
+
         # Standard employee cannot create and edit templates with dynamic inline fields
         with self.assertRaises(AccessError):
             self.env['mail.template'].with_user(self.user_employee).create({'email_to': '{{ object.partner_id.email }}'})
 
+        # Standard employee cannot edit his own templates if dynamic
+        with self.assertRaises(AccessError):
+            employee_template.with_user(self.user_employee).body_html = '<p t-esc="\'foo\'"></p>'
+
+        with self.assertRaises(AccessError):
+            employee_template.with_user(self.user_employee).email_to = '{{ object.partner_id.email }}'
+
+    def test_mail_template_acl_translation(self):
+        ''' Test that a user that doenn't have the group_mail_template_editor cannot create / edit
+        translation with dynamic code if he cannot write dynamic code on the related record itself.
+        '''
+
+        self.env.ref('base.lang_fr').sudo().active = True
+
+        employee_template = self.env['mail.template'].with_user(self.user_employee).create({'body_html': '<p>foo</p>'})
+
+        Translation = self.env['ir.translation']
+        Translation.insert_missing(employee_template._fields['body_html'], employee_template)
+        employee_translations = Translation.with_user(self.user_employee).search(
+            [('res_id', '=', employee_template.id), ('name', '=', 'mail.template,body_html'), ('lang', '=', 'fr_FR')]
+        )
+        # keep a copy to create new translation later
+        create_vals = employee_translations[0].read([])[0]
+
+        # write on translation for template without dynamic code is allowed
+        employee_translations[0].value = 'non-qweb'
```

Comment:
we could probably remove the [0] (maybe add a limit=1 in the search but that is probably not necessary)